### PR TITLE
[RELEASE] CVS-95549 - Hierarchical classification training failed without obvious reason

### PR DESCRIPTION
**Pull request by JihwanEom**
- Bugfix for drop_last was set to True in an invalid situation. (`self._patch_datasets` didn't parse hparams in template.yaml)

### CODE REVIEW

**Comment by JihwanEom** [comment]
run ote-test

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py [hunk 1]
```diff
@@ -287,11 +287,11 @@ def _init_recipe(self):
             logger.warning(f'train type {train_type} is not implemented yet.')
 
         self._recipe_cfg = MPAConfig.fromfile(recipe)
-        self._patch_datasets(self._recipe_cfg)  # for OTE compatibility
         self._patch_evaluation(self._recipe_cfg)  # for OTE compatibility
         logger.info(f'initialized recipe = {recipe}')
 
     def _init_model_cfg(self):
+        self._patch_datasets(self._recipe_cfg)  # for OTE compatibility, patch after overwriting hyperparams
```
Comment (goodsong81) [inline external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py]: Is this 'init_model' method the right place for 'patch_dataset'?
@harimkang Could you recommend another place to put?

**Review by goodsong81 — CHANGES_REQUESTED** [review]

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py (hunk 1 above)
Comment (harimkang) [reply external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py]: Doing `patch_datasets` inside the `init_model_cfg` function doesn't seem appropriate. It seems to affect tasks that do not require datasets. + need to check CLI tests for classification

Comment (supersoob) [reply external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py]: I saw drop_true is determined by batch size in _patch_datasets. I think this code part should be inserted somewhere after loading init_hparam, not in _patch_datasets.

Comment (JihwanEom) [reply external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py]: Reverted patch_datasets and moved to base_task.py. Please note cfg.drop_last will be parsed only in classification task, not for detection/segmentation task.

**Comment by JihwanEom** [comment]
run ote-test

**Comment by harimkang** [comment]
Hi @JihwanEom, new commit looks good. Could you run the classification api & cli test **locally** and share the results? You just need to do the classification. (Since it's a release branch, we need to carefully check the results with our tests.)

**Review by goodsong81 — APPROVED** [review]
LGTM.
But we need to refine the level of detail in the _run_task() code block and also other function. (e.g. which is in _initialize() but which is not)

**Comment by JihwanEom** [comment]
Thank you for detailed comments, here are CI build links:
https://ci-ote.iotg.sclab.intel.com/job/ote/job/pr-ote-test/676/
http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=pr-ote-test-676&environment=iotg-dev-workstation-08
There was no side effect by this PR based on validation reports.

**Review by harimkang — APPROVED** [review]
LGTM, I mistook him for running the CI test with a previous commit. (But it's run on a new commit)

**Pull request merged**